geop/src/main/scala/com/azavea/usace/programanalysis/geop/GeopServiceActor.scala

```diff
@@ -58,7 +76,30 @@ class GeopServiceActor(sc: SparkContext) extends Actor with HttpService {
                   .map { case (keys, count) =>
                     keys.mkString(",") -> JsNumber(count)
                   })
-              }).toVector
+            }).toVector
+          }
+        }
+      }
+    }
+
+  def tilesHandler =
+    pathPrefix(IntNumber / IntNumber / IntNumber) { (zoom, x, y) =>
+      respondWithMediaType(MediaTypes.`image/png`) {
+        complete {
+          future {
+            val result: Option[Tile] =
+              try {
+                Some(LayerReader(zoom, x, y, sc))
+              } catch {
+                case _: Exception => None
```

Comment:
Instead of catching `Exception`, consider being more specific like this:

```diff
@@ -18,7 +16,7 @@ import geotrellis.proj4.{LatLng, ConusAlbers}
 import geotrellis.raster.{IntConstantNoDataCellType, Tile}
 import geotrellis.raster.render.{IntColorMap, Exact}
 import geotrellis.raster.render.ColorMap.Options
-import geotrellis.spark.io.{TileNotFoundError}
+import geotrellis.spark.io.{AttributeNotFoundError, TileNotFoundError}


 class GeopServiceActor(sc: SparkContext) extends Actor with HttpService {
@@ -93,7 +91,7 @@ class GeopServiceActor(sc: SparkContext) extends Actor with HttpService {
               try {
                 Some(LayerReader(zoom, x, y, sc))
               } catch {
-                case _: Exception => None
+                case _: AttributeNotFoundError | _: TileNotFoundError => None
               }

             result.map { tile =>
```